dasharo-performance/pcie-power-management.robot

```diff
@@ -0,0 +1,44 @@
+*** Settings ***
+Library             Telnet    timeout=20 seconds    connection_timeout=120 seconds
+Library             SSHLibrary    timeout=90 seconds
+Library             RequestsLibrary
+Resource            ../rtectrl-rest-api/rtectrl.robot
+Resource            ../variables.robot
+Resource            ../keywords.robot
+
+Suite Setup         Prepare Test Suite
+Suite Teardown      Log Out And Close Connection
+
+
+*** Test Cases ***
+ASPM001.001 Check if disabling "Enable PCIe power management features" disables ASPM L0s
+    [Documentation]    Checks whether disabling PCIe power management features disables L0S state
+    Skip If    not ${APU_CONFIGURATION_MENU_SUPPORT}    APU configuration tests not supported.
+    Power On
+    ${setup_menu}=    Enter Setup Menu Tianocore And Return Construction
+    ${apu_menu}=    Enter Dasharo Submenu    ${setup_menu}    Dasharo APU Configuration
+    Set Option State    ${apu_menu}    Enable PCI Express power    ${FALSE}
+    Save Changes And Reset
+    Boot System Or From Connected Disk    ubuntu
+    Login To Linux
+    Switch To Root User
+    ${aspm_check}=    Execute Command In Terminal
+    ...    echo -n `lspci -vv | grep L0s | wc -l`
```

Comment:
Check for L1 too. Either as a separate test case or in this test case (the test title should also be updated then)

Reply:
Also this command will return multiple results. Also for PCI devices which we do not need to check ASPM.

We should only check the 00:02.1, 00:02.2, 00:02.3, 00:02.4 and 00:02.5 Root ports with their downstream devices. Also there is Dev Cap and Dev Ctl. We only care about Dev Ctl where there are the enable bits, not Dev Cap, where the ASPM or CLock PM capability is reported.

Reply:
L1 is not checked for because it's always enabled.

Reply:
And it shouldn't be enabled. Test would and should fail in such case

Reply:
OK, then let's leave this PR open and I will create a new issue. I don't want to write a test for L1 now, because I need some conditions to make sure the new test will work correctly, so the firmware must also first work correctly in order to write that test.

Reply:
> And it shouldn't be enabled. Test would and should fail in such case

So we should expect firmware fixes from you @miczyg1 in this regard?

Reply:
New issue at https://github.com/Dasharo/open-source-firmware-validation/issues/242

Reply:
I think this is one can be resolved @miczyg1 ?

Reply:
Yes